src/js/base/js-numbers.js

```diff
@@ -1201,16 +1201,44 @@ define("pyret-base/js/js-numbers", function() {
       return bnRemainder.call(m, n);
     });
 
+  var splitIntintoMantissaExpt = function(s) {
+    var str = s.toString();
+    var c0 = str[0];
+    var sign = '';
+
+
+    if (c0 === '-' || c0 === '+') {
+      str = str.substring(1);
+      if (c0 === '-') {
+        sign = '-';
+      }
+    }
+
+    var d0 = str[0];
+    str = str.substring(1);
+
+    var mantissa = Number(sign + d0 + '.' + str);
```

Comment:
I've added comments for both `splitIntIntoMantissaExpt` (was using English titlecase that keeps prepositions lowercase) and `_integerDivideToFixnum`. Yes, `split...` takes an arbitrarily precise int and returns a [mantissa, exponent] with generally some loss of precision, as both mantissa and exponent are constrained to be JS doubles. In addition, mantissa is in the interval [1, 10).